backend/src/v2/driver/driver.go

```diff
@@ -742,3 +742,19 @@ func validateVolumeMounts(podSpec *k8score.PodSpec) error {
 
 	return nil
 }
+
+func localPathForURI(uri string) (string, error) {
+	if strings.HasPrefix(uri, "gs://") {
+		return "/gcs/" + strings.TrimPrefix(uri, "gs://"), nil
+	}
+	if strings.HasPrefix(uri, "minio://") {
+		return "/minio/" + strings.TrimPrefix(uri, "minio://"), nil
+	}
+	if strings.HasPrefix(uri, "s3://") {
+		return "/s3/" + strings.TrimPrefix(uri, "s3://"), nil
+	}
+	if strings.HasPrefix(uri, "oci://") {
+		return "/oci/" + strings.ReplaceAll(strings.TrimPrefix(uri, "oci://"), "/", "_") + "/models", nil
+	}
+	return "", fmt.Errorf("failed to generate local path for URI %s: unsupported storage scheme", uri)
+}
```

Comment:
This duplicates the logic from `backend/src/v2/component/launcher_v2.go`.
Can we keep the `component.LocalPathForURI` function and use that in [retrieveArtifactPath](https://github.com/alyssacgoins/data-science-pipelines/blob/6e0c42c839fb4e499f3ae9374914305e407f03a7/backend/src/v2/component/launcher_v2.go#L992) (in `else`)? With that, we can revert all the changes in this file.